src/layout/Payment/queries/usePaymentInformationQuery.tsx

```diff
@@ -0,0 +1,23 @@
+import { useQuery } from '@tanstack/react-query';
+
+import { useAppQueries } from 'src/core/contexts/AppQueriesProvider';
+import { useProcessNavigation } from 'src/features/instance/ProcessNavigationContext';
+import { PaymentStatus } from 'src/layout/Payment/queries/types';
+
+export const usePaymentInformationQuery = (partyId?: string, instanceGuid?: string) => {
+  const { fetchPaymentInformation } = useAppQueries();
+  const { next } = useProcessNavigation() || {};
+  return useQuery({
+    queryKey: ['fetchPaymentInfo', partyId, instanceGuid],
+    queryFn: async () => {
+      if (partyId && instanceGuid) {
+        const result = await fetchPaymentInformation(partyId, instanceGuid);
+        if (result?.status === PaymentStatus.Paid) {
+          next && next({ action: 'confirm', nodeId: 'next-button' });
+        }
```

Comment:
Yikes - a get query that sometimes fires a mutation effect.. So if someone tries to render the `Payment` component with `renderAsSummary: true` in any other task type, this will end up calling `process/next` for them automatically? I would move this out from the query function and have that as an effect in the `Payment` component.